backend/devices/bitbox02/keystore.go

```diff
@@ -399,38 +414,55 @@ func (keystore *keystore) signBTCTransaction(btcProposedTx *btc.ProposedTransact
 			txOut.PkScript,
 		)
 
-		// outputAccountAddress represents the same address as outputAddress, but embeds the account configuration.
+		// outputAddress represents the same address as outputAddress, but embeds the account configuration.
 		// It is nil if the address is external.
-		outputAccountAddress := btcProposedTx.GetAccountAddress(blockchain.NewScriptHashHex(txOut.PkScript))
+		outputAddress, sameAccount, err := btcProposedTx.GetKeystoreAddress(btcProposedTx.SendingAccount, blockchain.NewScriptHashHex(txOut.PkScript))
+		if err != nil {
+			return errp.Newf("failed to get address: %v", err)
+		}
 
-		isOurs := outputAccountAddress != nil
+		isOurs := outputAddress != nil
 		if !isChange && !keystore.device.Version().AtLeast(semver.NewSemVer(9, 15, 0)) {
 			// For firmware older than 9.15.0, non-change outputs cannot be marked internal.
 			isOurs = false
 		}
 
 		var keypath []uint32
 		var scriptConfigIndex int
+		var outputScriptIndex *uint32
 		if isOurs {
-			keypath = outputAccountAddress.Configuration.AbsoluteKeypath().ToUInt32()
-			accountConfiguration := outputAccountAddress.AccountConfiguration
+			keypath = outputAddress.Configuration.AbsoluteKeypath().ToUInt32()
```

Comment:
Best not to supply this to the BitBox if `isOurs` is false, for clarity. I am a little surprised even that the BitBox doesn't error if this is supplied even if `ours` is false :yum:

Reply:
Yeah good idea (but no, it doesn't error :P) done